crates/runtime/src/datafusion/query/cache.rs

```diff
@@ -80,28 +105,34 @@ impl Query {
         parameters: Option<ParamValues>,
         tracker: Option<QueryTracker>,
     ) -> super::Result<PlanOrCached> {
-        // Try to get cached results first from sql
         let sql_cache_key = CacheKey::Query(sql, parameters.as_ref());
-        let (tracker, cache_status, sql_raw_cache_key) = match Self::try_get_cached_result(
+        let sql_or_user_cache_key = match request_context.client_supplied_cache_key() {
+            Some(user_key) => CacheKey::ClientSupplied(user_key),
+            _ => sql_cache_key,
```

Comment:
Should we move `CacheKey::Query(sql, parameters.as_ref());` from line 108 here?

Reply:
I see - there is special logic `// Always use CacheKey::Query when checking the plan cache`